public/404.html

```diff
@@ -1,67 +1,213 @@
+        width: 62%;
+      }
+    }
+
+    @media(max-width:1024px){
+      .notfound-top {
+        padding: 0em 0em 0em 0em;
+      }
+      .content h3 {
+        margin: 10px 0px;
+      }
+      .notfound-top h1 {
+        font-size: 8em;
+        padding-top: 0.35em;
+      }
+    }
+
+    @media(max-width:768px){
+      .notfound-top h1 {
+        font-size: 7.5em;
+      }
+      .content {
+        width: 52%;
+        padding-left: 2em;
+      }
+      .not-found-container {
+        width: 78%;
+        padding: 5em 3em 5em 3em;
+        margin: 12em auto 0em;
+      }
+      .content h3 {
+        font-size: 13px;
+      }
+    }
+
+    @media(max-width:667px){
+      .notfound-top h1 {
+        font-size: 6.5em;
+        padding-top: 0.5em;
+      }
+    }
+
+    @media(max-width:640px){
+      .notfound-top {
+        float: none;
+        width: 34%;
+        margin: 0 auto;
+      }
+      .content {
+        width: 100%;
+        float: none;
+        margin: 0 auto;
+        padding-left: 0em;
+        text-align: center;
+        border-left: none;
+      }
+      .notfound-top h1 {
+        padding-top: 0em;
+      }
+      .not-found-container {
+        width: 78%;
+        padding: 3em 3em 3em 3em;
+        margin: 7em auto 0em;
+      }
+    }
+
+    @media(max-width:600px){
+      .notfound-top {
+        width: 39%;
+      }
+    }
+
+    @media(max-width:568px){
+      .notfound-top {
+        width: 42%;
+      }
+    }
+
+    @media(max-width:480px){
+      .not-found-container {
+        width: 70%;
+        margin: 6em auto 0em;
+      }
+      .notfound-top h1 {
+        font-size: 5em;
+      }
+      .content img {
+        width: 20%;
+      }
+      .notfound-top {
+        width: 41%;
+      }
+    }
+
+    @media(max-width:414px){
+      .notfound-top {
+        width: 50%;
+      }
+    }
+
+    @media(max-width:384px){
+      .notfound-top {
+        width: 53%;
+      }
+      .not-found-container {
+        padding: 2em 3em 2em 3em;
+      }
+    }
+
+    @media(max-width:320px){
+      .not-found-container {
+        width: 83%;
+        margin: 3em auto 0em;
+        padding: 1em 1em 1em 1em;
+      }
+      .notfound-top {
+        width: 44%;
+      }
+      .notfound-top h1 {
+        font-size: 4em;
+      }
+      .content h3 {
+        margin: 10px 0px;
+        font-size: 12px;
+      }
+    }
   </style>
 </head>
 
 <body>
-  <!-- This file lives in public/404.html -->
-  <div class="dialog">
-    <div>
-      <h1>The page you were looking for doesn't exist.</h1>
-      <p>You may have mistyped the address or the page may have moved.</p>
+  <div class="not-found-container">
+    <div class="notfound-top">
+      <h1>404</h1>
+    </div>
+    <div class="content">
+      <h2>Github Classroom</h2>
+      <h4>Page not Found</h4>
+      <h3>The page you are trying to reach does not exist, or has been moved.</h3>
+      <h4><a href="https://classroom.github.com/">Get back to Classroom</a></h4>
```

Comment:
> Can we change this to Back to Classroom or Go back to Classroom?

I'll change it to Go back to Classroom

> In general, we try to avoid hard-coding URLs in rails, since hosts can change easily. Especially in this case where our users may be running their own instance of GitHub classroom on their own server. Bringing them to our hosted version would be quite confusing.

Agree! But pages `404.html` and `invalid_link_error.html` are pure html pages. We can't add `link_to` unless they have `.erb`. I will try n search if there is a workaround to insert root_url instead of GitHub classroom link. If you have any will be glad to implement it.